x-pack/filebeat/input/httpjson/request.go

```diff
@@ -163,21 +181,160 @@ func newRequester(
 	}
 }
 
-func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
-	req, err := r.requestFactory.newHTTPRequest(stdCtx, trCtx)
+// collectResponse returns response from provided request
+func (rf *requestFactory) collectResponse(stdCtx context.Context, trCtx *transformContext, r *requester, publisher inputcursor.Publisher) (*http.Response, error) {
+	req, err := rf.newHTTPRequest(stdCtx, trCtx)
 	if err != nil {
-		return fmt.Errorf("failed to create http request: %w", err)
+		return nil, fmt.Errorf("failed to create http request: %w", err)
 	}
-
 	httpResp, err := r.client.do(stdCtx, trCtx, req)
 	if err != nil {
-		return fmt.Errorf("failed to execute http client.Do: %w", err)
+		return nil, fmt.Errorf("failed to execute http client.Do: %w", err)
 	}
-	defer httpResp.Body.Close()
+	return httpResp, nil
+}
 
-	eventsCh, err := r.responseProcessor.startProcessing(stdCtx, trCtx, httpResp)
+// generateNewUrl returns new url value using replacement from oldUrl with str ids
+func generateNewUrl(replacement string, oldUrl string, str []byte) (url.URL, error) {
+	reg, err := regexp.Compile(replacement)
+	if err != nil {
+		return url.URL{}, fmt.Errorf("failed to create regex on provided value: %w", err)
+	}
+	newUrl, err := url.Parse(string(reg.ReplaceAll([]byte(oldUrl), str)))
 	if err != nil {
-		return err
+		return url.URL{}, fmt.Errorf("failed to replace value in url: %w", err)
+	}
+	return *newUrl, nil
+
+}
+
+func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
+	var (
+		n                 int
+		ids               []string
+		err               error
+		split             *split
+		urlCopy           url.URL
+		urlString         string
+		httpResp          *http.Response
+		intermediateResps []*http.Response
+		finalResps        []*http.Response
+	)
+	for i, rf := range r.requestFactory {
+		// iterate over collected ids from last response
+		if i != 0 {
```

Comment:
What about inverting the logic here so the first block of code is also the first to be executed? I believe it will improve readability.
```suggestion
		if i == 0 {
```